### ISSUE TITLE
Support query shift on FEMNIST

### ISSUE THREAD

**Issue by r09921064** [ebennequin/meta-domain-shift#25]
Dear author,

I've gone through the code and met some problems.
Distribution shifts seem to be presented with digits written by different writers in FEMNIST,
I'm wondering how to turn off the shift since the number of instance of a digit from one writer is small.
Furthermore, in evaluation_config.py, the parameter SUPPORT_QUERY_SHIFT seems not working to FEMNIST.
I will be grateful for any help you can provide, thanks!

Best Regards,
Wei Ding

**Comment by ebennequin** [comment]
Hi!

Thanks for your very useful feedback. It seems that we made a mistake for this experiment: we didn't version the code we used to generate the results for FEMNIST in the "no shift" setting, and the results were incorrect.

I just merged [some new code](https://github.com/ebennequin/meta-domain-shift/pull/26) allowing to reproduce this experiment. Running the experiment with the seeds 1, 2 and 3, we obtained **99.72% +- 0.07** (which is way more consistent than the previously claimed results: intuitively it's very easy to recognize a digit written by the same person).

We will update [the article on arxiv](https://arxiv.org/abs/2105.11804) accordingly in the next few days.

Note that for our experiments we only turn off the support-query shift for evaluation. Since the test set of FEMNIST is only composed of digits (no letters) which have several instances for each writer, it's not that hard to build episodes with no shift on the test set (it would be harder on the train set or validation set).

I am very sorry about this mistake, and I hope this helped.
Best regards,
Etienne
